middleware/proxy/reverseproxy.go

```diff
@@ -153,8 +155,37 @@ func (p *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request, extr
 
 	copyHeader(rw.Header(), res.Header)
 
-	rw.WriteHeader(res.StatusCode)
-	p.copyResponse(rw, res.Body)
+	if res.StatusCode == HTTPSwitchingProtocols && outreq.Header.Get("Upgrade") == "websocket" {
```

Comment:
So, the client making the original HTTP request with the Upgrade header will get back a 200 status code and a response without any Upgrade header. Is that expected, for a websocket reverse proxy? Or should it forward the 101 status and the Upgrade header back down to the client?